src/test/kotlin/reminder/application/ReminderSearcherTest.kt

```diff
@@ -0,0 +1,60 @@
+package reminder.application
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import reminder.domain.Reminder
+import reminder.domain.ReminderIdentifierGenerator
+import reminder.domain.ReminderRepository
+import reminder.mothers.ReminderMother
+import java.util.*
+import kotlin.test.assertEquals
+
+class ReminderSearcherTest {
+    private lateinit var reminderSaver: ReminderSaver
+    private lateinit var generator: ReminderIdentifierGenerator
+    private lateinit var reminderSearcher: ReminderSearcher
+    private lateinit var repository: ReminderRepository
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(ReminderRepository::class.java)
+        generator = Mockito.mock(ReminderIdentifierGenerator::class.java)
+        reminderSearcher = ReminderSearcher(repository)
+        reminderSaver = ReminderSaver(repository, generator)
+    }
+
+    @Test
+    fun `Nothing is returned when valid identifier doesn't exist in the repository`() {
+        val reminder = ReminderMother.getValidReminderWithoutDescription()
+        `Assert that the reminder was not found in the repository given the identifier from`(reminder)
+    }
+
+    @Test
+    fun `Reminder is returned when identifier exists in the repository`() {
+        val reminder = ReminderMother.getValidReminderWithoutDescription()
+        `Save reminder to repository given the primitives from`(reminder)
```

Comment:
You don't really have to save the repository, you can only say to mockito that a call to the function will return the reminder